clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3788,18 +4446,36 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     return true;
   }
 
+  bool handleSyclSpecialType(ParmVarDecl *, QualType) final {
+    // TODO
+    unsupportedFreeFunctionParamType();
+    return true;
+  }
+
   bool handlePointerType(FieldDecl *FD, QualType FieldTy) final {
     addParam(FD, FieldTy,
              ((StructDepth) ? SYCLIntegrationHeader::kind_std_layout
                             : SYCLIntegrationHeader::kind_pointer));
     return true;
   }
 
+  bool handlePointerType(ParmVarDecl *PD, QualType ParamTy) final {
+    addParam(PD, ParamTy,
```

Comment:
Please simplify this for your PR. You just need to addParam with kind_pointer

Reply:
Yes, changed.